jerry-core/ecma/builtin-objects/typedarray/ecma-builtin-typedarray-prototype.c

```diff
@@ -988,6 +989,94 @@ ecma_builtin_typedarray_prototype_object_to_string (ecma_value_t this_arg) /**<
   return ret_value;
 } /* ecma_builtin_typedarray_prototype_object_to_string */
 
+/**
+ * The %TypedArray%.prototype object's 'subarray' routine.
+ *
+ * See also:
+ *          ES2015, 22.2.3.26
+ *
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value.
+ */
+static ecma_value_t
+ecma_builtin_typedarray_prototype_subarray (ecma_value_t this_arg, /**< this argument */
+                                            ecma_value_t begin, /**< begin */
+                                            ecma_value_t end) /**< end */
+{
+  ecma_value_t ret_value = ECMA_VALUE_EMPTY;
+
+  /* 2.~ 4. */
+  if (!ecma_is_typedarray (this_arg))
+  {
+    return ecma_raise_type_error (ECMA_ERR_MSG ("Argument 'this' is not a TypedArray."));
+  }
+
+  ecma_object_t *src_typedarray_p = ecma_get_object_from_value (this_arg);
+
+  /* 5. buffer */
+  ecma_object_t *src_typedarray_arraybuffer_p = ecma_typedarray_get_arraybuffer (src_typedarray_p);
+
+  /* 6. srcLength */
+  ecma_length_t src_length = ecma_typedarray_get_length (src_typedarray_p);
+
+  /* 9. beginIndex, 12. endIndex */
+  uint32_t begin_index_uint32 = 0, end_index_uint32 = 0;
+
+  /* 7. relativeBegin */
+  ECMA_OP_TO_NUMBER_TRY_CATCH (relative_begin, begin, ret_value);
+  begin_index_uint32 = ecma_builtin_helper_array_index_normalize (relative_begin, src_length);
+
+  if (ecma_is_value_undefined (end))
+  {
+    end_index_uint32 = (uint32_t) src_length;
+  }
+  else
+  {
+    /* 10. relativeEnd */
+    ECMA_OP_TO_NUMBER_TRY_CATCH (relative_end, end, ret_value);
+
+    end_index_uint32 = ecma_builtin_helper_array_index_normalize (relative_end, src_length);
+
+    ECMA_OP_TO_NUMBER_FINALIZE (relative_end);
+  }
+
+  ECMA_OP_TO_NUMBER_FINALIZE (relative_begin);
+
+  /* 13. newLength */
+  ecma_length_t subarray_length = 0;
+  if ((int64_t) end_index_uint32 - begin_index_uint32 > 0)
+  {
+    subarray_length = end_index_uint32 - begin_index_uint32;
+  }
```

Comment:
I think the `int64_t` cast can be eliminated like this:
```
  ecma_length_t subarray_length = 0;
  if (end_index_uint32 > begin_index_uint32)
  {
    // Cannot underflow
    subarray_length = end_index_uint32 - begin_index_uint32;
  }
  // Otherwise the subarray_length must be 0
```